### PR TITLE
PAT-136: Adding in a callback prop, allowing a callback to be invoked when accordion is toggled

### PR DESCRIPTION
**Pull request by rohannair**

### CODE REVIEW

File: src/accordion/accordion-header.js [hunk 1]
```diff
@@ -11,15 +11,19 @@ class AccordionHeader extends Component {
...
 
-  _toggleContent = () => {
+
+  _onClick = () => {
     if (this.props.isEnabled) {
       this.props.toggleOne(this.props.index);
     }
+
+    // Invoke any callbacks on the toggle
+    if (this.props.cb) this.props.cb(e, this.props);
```
Comment (maedhr) [inline src/accordion/accordion-header.js]: Should this be in the above if block?

File: src/accordion/accordion-header.js [hunk 2]
```diff
@@ -4,22 +4,28 @@ import classNames from 'classnames';
...
   }
 
-  _toggleContent = () => {
+
+  _onClick = (e) => {
     if (this.props.isEnabled) {
       this.props.toggleOne(this.props.index);
+
+      // Invoke any callbacks on the toggle
+      if (this.props.cb) this.props.cb(e, this.props);
```
Comment (SeeThruHead) [inline src/accordion/accordion-header.js]: Can we use something more descrptive than `callback` and `cb`

File: src/accordion/index.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ class Accordion extends Component {
     sections:            PropTypes.array.isRequired,
     uniqueIdentifier:    PropTypes.string,
     initialSectionIndex: validateIndex,
-    openSectionIndex:    validateIndex
+    openSectionIndex:    validateIndex,
```
Comment (SeeThruHead) [inline src/accordion/index.js]: comma dangle

Comment (rohannair) [inline src/accordion/index.js]: Dealt with by Babel. This is on purpose actually, as it means easier reordering of properties.

File: src/accordion/accordion-header.js (hunk 2 above)
Comment (rohannair) [inline src/accordion/accordion-header.js]: Yes, but it is literally just a callback.

**Pull request merged**